### PR TITLE
Disable `git update-git-for-windows`

### PR DESCRIPTION
**Pull request by dscho**
This addresses https://github.com/microsoft/git/issues/241

### CODE REVIEW

**Review by derrickstolee — APPROVED** [review]
Nice! So by making `git update` an actual builtin, the script version from git-for-windows/build-extra will never be visible? That seems like a good approach.

File: help.c [hunk 1]
```diff
@@ -794,3 +794,19 @@ NORETURN void help_unknown_ref(const char *ref, const char *cmd,
...
+		   "\t(disabled in VFSforGit)"),
+		NULL
+	};
+	struct option options[] = {
+		OPT_END()
+	};
+
+	argc = parse_options(argc, argv, prefix, options, usage, 0);
+
+	die(_("git %s is not supported in VFSforGit"), argv[0]);
```
Comment (jrbriggs) [inline help.c:811]: nit: this version of Git ships in both VFS for Git and Scalar. Can we make this less specific and say something like `not supported in this build of Git for Windows`?

**Review by jrbriggs — APPROVED** [review]

**Comment by dscho** [comment]
> by making `git update` an actual builtin, the script version from git-for-windows/build-extra will never be visible?

Yep ;-)

File: help.c (hunk 1 above)
Comment (dscho) [reply help.c:811]: Sure.

Comment (dscho) [reply help.c:811]: Done ;-)

**Pull request merged**